bip-tx-terminology.mediawiki

```diff
@@ -0,0 +1,324 @@
+The hope is that this terminology could be used in technical writing about Bitcoin as well as future BIPs.
+
+===Objects and their Serialization===
+
+We treat the objects of the Bitcoin protocol (e.g. transactions, blocks) as distinct from their serialization.
+We distinguish concrete properties of transactions such as fields and scripts (labeled [Component]) from serialization artifacts (labeled [Artifact]).
+For example, version and lock time are fields that can be set during transaction creation and therefore components of a transaction, while the input counter is a serialization artifact resulting from how the input list of the transaction is represented in the transfer format.
+Likewise, the input script is a transaction component, but the length indicator of the input script is a serialization artifact.
+
+We also introduce some umbrella terms, concepts, and ideas that are useful to describe aspects of transactions (labeled [Concept]).
+
+===Anatomy of a serialized transaction===
+
+In this section we dissect a serialized transaction. Each sub-level in the list repeats a subset of the transaction’s serialization to label the included parts. The goal is to provide a map from a serialized transaction back to our definitions above. We use the first ever P2TR transaction with a keypath and a scriptpath input <code>37777defed8717c581b4c0509329550e344bdc14ac38f71fc050096887e535c8</code>.
+
+* Raw '''Transaction''' <code>020000000001027bc0bba407bc67178f100e352bf6e047fae4cbf960d783586cb5e430b3b700e70000000000feffffff7bc0bba407bc67178f100e352bf6e047fae4cbf960d783586cb5e430b3b700e70100000000feffffff01b4ba0e0000000000160014173fd310e9db2c7e9550ce0f03f1e6c01d833aa90140134896c42cd95680b048845847c8054756861ffab7d4abab72f6508d67d1ec0c590287ec2161dd7884983286e1cd56ce65c08a24ee0476ede92678a93b1b180c03407b5d614a4610bf9196775791fcc589597ca066dcd10048e004cd4c7341bb4bb90cee4705192f3f7db524e8067a5222c7f09baf29ef6b805b8327ecd1e5ab83ca2220f5b059b9a72298ccbefff59d9b943f7e0fc91d8a3b944a95e7b6390cc99eb5f4ac41c0d9dfdf0fe3c83e9870095d67fff59a8056dad28c6dfb944bb71cf64b90ace9a7776b22a1185fb2dc9524f6b178e2693189bf01655d7f38f043923668dc5af45bffd30a00</code>
+** First part of '''Transaction Header'''
+*** '''Transaction Version''' <code>02000000</code>: integer field, here indicating version 2
+*** '''Marker''' <code>00</code>: serialization artifact indicating extended serialization
+*** '''Flag''' <code>01</code>: serialization artifact indicating presence of a witness structure
+** '''Transaction Input List'''
+*** Length of '''Transaction Input List''' <code>02</code>: serialization artifact, <code>CompactSize</code> here indicating 2 inputs, considered part of the transaction header
+*** First '''Input'''
+**** '''Outpoint'''
+***** '''txid''' <code>7bc0bba407bc67178f100e352bf6e047fae4cbf960d783586cb5e430b3b700e7</code>: little-endian txhash indicating that the spent TXO was created by the transaction e700b7b330e4b56c5883d760f9cbe4fa47e0f62b350e108f1767bc07a4bbc07b.
```

Comment:
pedantic mode ad "little-endian": a txid is a hash result and hence just a sequence of bytes, the concept of endianness only becomes relevant if we'd ever want to treat it as integer (which we do for block-hashes w.r.t. PoW calculations, but AFAIK never for txids). It's just that for some reason the txids are all _shown_ in reverse when presented to / parsed from the user (see uint256.GetHex()/SetHex()), but on the wire the txid is serialized in the same order as the double-sha256 hash result. So in my world-view, if we want to keep the endianness terminology, the way we store txids internally and on the wire would be "big-endian", and "little-endian" is used for displaying and parsing them. Wrote about other unfortunate consequences of that need for txid reversing a longer time ago here: https://github.com/bitcoin/bitcoin/pull/24952#issuecomment-1108794823.

I'd also mildly agree to the anwer in https://bitcoin.stackexchange.com/questions/2063/why-does-the-bitcoin-protocol-use-the-little-endian-notation/2069#2069: _"Hashes are defined by the standards as being big-endian, and crypto libraries deal with them in that form, so hashes are transmitted in big-endian. Bitcoin displays hashes in little-endian because Bitcoin sometimes considers hashes to be little-endian integers instead of strings."_ (that "sometimes" seems to be only for block hashes, as far as I'm aware)